raiden/tests/fuzz/test_state_changes.py

```diff
@@ -524,136 +584,169 @@ def _new_mediator_transfer(
         action = ReceiveSecretReveal(secret, sender)
-        result = node.state_transition(self.chain_state, action)
+        result = node.state_transition(client.chain_state, action)
         assert not result.events
 
     # pylint: disable=no-value-for-parameter
     @rule(
-        target=secret_requests, previous_action=consumes(init_mediators), invalid_sender=address()
+        target=secret_requests,
+        previous_action_with_address=consumes(init_mediators),
+        invalid_sender=address(),
     )
     # pylint: enable=no-value-for-parameter
-    def wrong_address_receive_secret_reveal(self, previous_action, invalid_sender):
+    def wrong_address_receive_secret_reveal(self, previous_action_with_address, invalid_sender):
+        previous_action, client, our_address = self._unwrap(previous_action_with_address)
         secret = self.secrethash_to_secret[previous_action.from_transfer.lock.secrethash]
         invalid_action = ReceiveSecretReveal(secret, invalid_sender)
-        result = node.state_transition(self.chain_state, invalid_action)
+        result = node.state_transition(client.chain_state, invalid_action)
         assert not result.events
 
         valid_sender = previous_action.from_transfer.target
         valid_action = ReceiveSecretReveal(secret, valid_sender)
-        return valid_action
+        return WithOurAddress(our_address=our_address, data=valid_action)
 
 
 class OnChainMixin:
 
-    block_number: BlockNumber
+    block_number: typing.BlockNumber
 
     @rule(number=integers(min_value=1, max_value=50))
     def new_blocks(self, number):
-        events = list()
-
         for _ in range(number):
             block_state_change = Block(
                 block_number=self.block_number + 1,
                 gas_limit=1,
                 block_hash=factories.make_keccak_hash(),
             )
-            result = node.state_transition(self.chain_state, block_state_change)
-            events.extend(result.events)
+            for client in self.address_to_client.values():
+                events = list()
+                result = node.state_transition(client.chain_state, block_state_change)
+                events.extend(result.events)
 
             self.block_number += 1
 
-    @rule(target=partners)
-    def open_channel(self):
-        return self.new_channel_with_transaction()
+    @rule(reference=address_pairs, target=address_pairs)
+    def open_channel(self, reference):
+        return self.new_channel_with_transaction(reference.our_address)
 
-    @rule(partner=consumes(partners))
-    def settle_channel(self, partner):
-        channel = self.address_to_channel[partner]
+    @rule(address_pair=consumes(address_pairs))
+    def settle_channel(self, address_pair):
+        our_address = address_pair.our_address
+        partner = address_pair.partner_address
+        client = self.address_to_client[our_address]
+        channel = client.address_to_channel[partner]
```

Comment:
My personal preference would be
```
client = self.address_to_client[address_pair.our_address]
channel = client.address_to_channel[address_pair.partner_address]
```
if the intermediate variables (`our_address` and `partner`) are not used anywhere else in the function and don't add any additional meaning.

Reply:
True, that looks better